### PR TITLE
Fixed receiving objects in chunk

### PR DESCRIPTION
**Pull request by samartnik**
This fixes https://github.com/brave/browser-android-tabs/issues/925

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
Tested locally with brave-core. This PR fixes https://github.com/brave/brave-browser/issues/2283 and also does huge speedup for syncing of large folders.

File: client/requestUtil.js [hunk 1]
```diff
@@ -173,13 +173,16 @@ RequestUtil.prototype.parseAWSResponse = function (bytes) {
...
  */
-RequestUtil.prototype.list = function (category, startAt, maxRecords) {
+RequestUtil.prototype.list = function (category, startAt, maxRecords, nextContinuationToken) {
   const prefix = `${this.apiVersion}/${this.userId}/${category}`
   let options = {
     MaxKeys: maxRecords || 1000,
     Bucket: this.bucket,
     Prefix: prefix
   }
+  if (nextContinuationToken != '') {
```
Comment (darkdh) [inline client/requestUtil.js]: lint: should be `!==`

File: lib/s3Helper.js [hunk 2]
```diff
@@ -221,7 +221,8 @@ module.exports.listObjects = function (s3, options, limitResponse) {
         } else {
           resolve({
             contents: data.Contents,
-            isTruncated: data.IsTruncated
+            isTruncated: data.IsTruncated,
+            nextContinuationToken : data.NextContinuationToken
```
Comment (darkdh) [inline lib/s3Helper.js]: lint: extra space before colon

File: lib/s3Helper.js [hunk 3]
```diff
@@ -230,7 +231,8 @@ module.exports.listObjects = function (s3, options, limitResponse) {
         if (error) { reject(error) }
         resolve({
           contents: data,
-          isTruncated: false
+          isTruncated: false,
+          nextContinuationToken : ''
```
Comment (darkdh) [inline lib/s3Helper.js]: lint: extra space before colon

**Pull request merged**